src/Index.hpp

```diff
@@ -110,6 +110,131 @@ namespace nbs {
             return types;
         }
 
+
+        // Get the timestamp where as many subtypes move forward without moving forward twice (still allowed when passed
+        // a single type)
+        std::uint64_t nextTimestamp(const uint64_t& timestamp,
+                                    const std::vector<TypeSubtype>& types,
+                                    const int& steps) {
+
+            auto matchingIndexItems = this->getIteratorsForTypes(types);
+
+            std::vector<TimestampState> states;
+
+            for (auto& iteratorPair : matchingIndexItems) {
+                auto& begin = iteratorPair.first;
+                auto& end   = iteratorPair.second;
+
+                IndexItemFile target;
+                target.item.timestamp = timestamp;
+
+
+                if (begin != end) {
+                    // Find the first item with a timestamp greater than the requested timestamp
+                    auto position =
+                        std::upper_bound(begin, end, target, [](const IndexItemFile& a, const IndexItemFile& b) {
+                            return a.item.timestamp < b.item.timestamp;
+                        });
+
+                    // This is before the beginning, the first packet is actually begin
+                    if (position != begin || position != end) {
```

Comment:
Shouldn't need the check for end, as previous for end will be the one before the end
```suggestion
                    if (position != begin) {
```

Reply:
Algorithm was refactored so this code segment is not used anymore.